### ISSUE TITLE
rvierdiiev - DelayedWETH.unlock extends existing lock

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-02-optimism-2024-judging#173]
rvierdiiev

medium

# DelayedWETH.unlock extends existing lock

## Summary
Each new DelayedWETH.unlock call extends already existing lock for the user, thus he should wait longer.

## Vulnerability Detail
DelayedWETH.unlock [is called by FaultDisputeGame](https://github.com/sherlock-audit/2024-02-optimism-2024/blob/main/optimism/packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol#L704), then [timestamp for user's lock is set to current time](https://github.com/sherlock-audit/2024-02-optimism-2024/blob/main/optimism/packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol#L64). This means that in case if user already had existing lock, then he [should to wait full DELAY_SECONDS period](https://github.com/sherlock-audit/2024-02-optimism-2024/blob/main/optimism/packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol#L79C32-L79C45) to withdraw.

DelayedWETH.unlock can be called several times during the game, when resolving claims. And claim that was resolved first, may be withdrawn only after last claim + DELAY_SECONDS. So if claim 1 for user was resolved at day1 and last claim was resolved on day 3, then user will be able to withdraw first claim funds at day1 + DELAY_SECONDS.

## Impact
User need to wait longer to withdraw
## Code Snippet
Provided above
## Tool used

Manual Review

## Recommendation
Make each withdrawal be tracked separately.

Duplicate of #195
